sync-github-ado.yml

```diff
@@ -0,0 +1,34 @@
+# variables:
+  # ADOUrl: defined on azure portal
+
+trigger:
+  batch: true
+  branches:
+    include:
+    - v3
+    - v3-release
+    - dev
+    - main
+
+variables:
+  NugetSecurityAnalysisWarningLevel: none
+  
+jobs:
+- job: 'Sync_GitHub_To_AzureDevOps'
+  pool: 
+    vmImage: 'windows-latest'
+  steps:
+  - task: Powershell@2
+    displayName: sync from GitHub to Azure DevOps
+    env:
+      SYSTEM_ACCESSTOKEN: $(System.AccessToken)
+    inputs:
+      targetType: inline
+      script: |
+        $branch = "$(Build.SourceBranch)"
+        git init
+        git fetch $(Build.Repository.Uri) $branch
+        git checkout FETCH_HEAD
+        git checkout -b $branch
```

Comment:
> Not really, you need to checkout the specific branch firstly

We could use `git push https://xyz HEAD:$branch`

Reply:
![image](https://user-images.githubusercontent.com/5196139/127089909-2e6a1feb-8ca2-4d34-a58a-57b2d9a11ce2.png)
Not able to push without upstream branch.